src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -102,14 +106,149 @@ public HashFileStore(int depth, int width, String algorithm, Path storeDirectory
         this.objectStoreAlgorithm = algorithm;
     }
 
+    @Override
+    public HashAddress storeObject(InputStream object, String pid, String additionalAlgorithm, String checksum,
+            String checksumAlgorithm)
+            throws NoSuchAlgorithmException, IOException, SecurityException, FileNotFoundException,
```

Comment:
not sure how useful/necessary all these are: there's some redundancy, since they are all subclasses of either `RuntimeException` or `IOException`, which are also listed here. I do like that they are all detailed in the javadoc for the interface, though. 

I'd be interested in @taojing2002's take this one - what makes most sense?

Reply:
I will be creating customized exception classes to reduce the redundancy here and throwing the new exception accordingly (in another development branch).